### PR TITLE
fix oc resources not apply for clean up data field in configmap or secret

### PR DESCRIPTION
**Pull request by BumbleFeng**
Fix https://gitlab.cee.redhat.com/service/app-interface/-/merge_requests/9999 not get apply

Signed-off-by: Feng Huang <fehuang@redhat.com>

### CODE REVIEW

**Review by maorfr — APPROVED** [review]
this is a nice fix for ConfigMaps, which is the main use case this bug encounters.
another use case we need to think about is label selectors.

LGTM
thank you!

File: utils/openshift_resource.py [hunk 1]
```diff
@@ -46,6 +48,10 @@ def obj_intersect_equal(self, obj1, obj2):
                     pass
                 elif self.ignorable_key_value_pair(obj1_k, obj1_v):
                     pass
+                elif obj1_k == 'data':
+                    diff = [k for k in obj2_v if k not in obj1_v]
+                    if len(diff) > 0:
```
Comment (apahim) [inline utils/openshift_resource.py:53]: From PEP8 "Programming Recommendations" (https://www.python.org/dev/peps/pep-0008/#id51):

```
For sequences, (strings, lists, tuples), use the fact that empty sequences are false:

# Correct:
if not seq:
if seq:

# Wrong:
if len(seq):
if not len(seq):
```

**Comment by BumbleFeng** [comment]
Add support for labels
This change will remove some data field for 
```
[2020-10-01 09:23:43] [INFO] [openshift_base.py:apply:204] - ['apply', 'app-sre-prod-04', 'uhc-production', 'ConfigMap', 'authentication']
[2020-10-01 09:23:43] [INFO] [openshift_base.py:apply:204] - ['apply', 'app-sre-stage-01', 'uhc-stage', 'ConfigMap', 'authentication']
[2020-10-01 09:23:43] [INFO] [openshift_base.py:apply:204] - ['apply', 'hive-production', 'gcp-project-operator', 'ConfigMap', 'gcp-project-operator']
```
And remove label for 
```
[2020-10-01 09:23:43] [INFO] [openshift_base.py:apply:204] - ['apply', 'app-sre-stage-01', 'openshift-customer-monitoring', 'ServiceMonitor', 'ams-subscription-labels']
```

**Comment by maorfr** [comment]
these 2 should not happen:
```
[2020-10-01 09:23:43] [INFO] [openshift_base.py:apply:204] - ['apply', 'app-sre-prod-04', 'uhc-production', 'ConfigMap', 'authentication']
[2020-10-01 09:23:43] [INFO] [openshift_base.py:apply:204] - ['apply', 'app-sre-stage-01', 'uhc-stage', 'ConfigMap', 'authentication']
```

since the data in these configmaps is automatically generated.
we should ignore it somehow.

**Review by maorfr — APPROVED** [review]
LGTM

**Pull request merged**